qupath-gui-fx/src/main/java/qupath/lib/gui/QuPathGUI.java

```diff
@@ -4078,6 +4100,9 @@ public boolean installTool(PathTool tool, KeyCodeCombination accelerator) {
 	 * @param tool
 	 */
 	public void setSelectedTool(PathTool tool) {
+		// Record which tools was currently selected
+		previousTool = getSelectedTool();
```

Comment:
I think this should be moved down to just above
```java
this.selectedToolProperty.set(tool);
```
so that it is not called if the tool doesn't change.

Also, should you explicitly avoid setting the previous tool if the `MoveTool` is selected? Reason: if I intentionally switch to the `MoveTool` I can end up locked in a scenario where pressing the middle button just toggles between the `MoveTool` being selected... with no obvious difference.

Reply:
Hi Pete,

> I think this should be moved down to just above
> `this.selectedToolProperty.set(tool);`

done, you're right!

> Also, should you explicitly avoid setting the previous tool if the `MoveTool` is selected?

Yes! I just implemented this, it should work now. Clicking the move button retains the previously selected tool.